packages/repository/test/unit/mixins/repository.mixin.unit.ts

```diff
@@ -141,35 +141,66 @@ describe('RepositoryMixin dataSource', () => {
     expect(withoutDataSource).to.be.empty();
   });
 
-  it('binds dataSource to a binding key using the dataSource name property', () => {
+  it('binds dataSource class using the dataSourceName property', () => {
     const myApp = new AppWithRepoMixin();
-    const fooDataSource: juggler.DataSource = new juggler.DataSource({
-      name: 'foo',
-      connector: 'memory',
-    });
-    myApp.dataSource(fooDataSource);
-    expectDataSourceToBeBound(myApp, fooDataSource, 'foo');
+
+    myApp.dataSource(FooDataSource);
+    expectDataSourceToBeBound(myApp, FooDataSource, 'foo');
+  });
+
+  it('binds dataSource class using the given name', () => {
+    const myApp = new AppWithRepoMixin();
+    myApp.dataSource(FooDataSource, 'bar');
+    expectDataSourceToBeBound(myApp, FooDataSource, 'bar');
   });
 
-  it('binds dataSource to a binding key using the given name', () => {
+  it('binds dataSource class using Class name', () => {
     const myApp = new AppWithRepoMixin();
-    const barDataSource: juggler.DataSource = new juggler.DataSource({
-      connector: 'memory',
-    });
-    myApp.dataSource(barDataSource, 'bar');
-    expectDataSourceToBeBound(myApp, barDataSource, 'bar');
+    myApp.dataSource(BarDataSource);
+    expectDataSourceToBeBound(myApp, BarDataSource, 'BarDataSource');
+  });
+
+  it('binds dataSource class instance using dataSourceName property', () => {
+    const myApp = new AppWithRepoMixin();
+    myApp.dataSource(new FooDataSource());
+    expectDataSourceToBeBound(myApp, FooDataSource, 'foo');
+  });
+
+  it('binds dataSource class instance using custom name', () => {
+    const myApp = new AppWithRepoMixin();
+    myApp.dataSource(new FooDataSource(), 'bar');
+    expectDataSourceToBeBound(myApp, FooDataSource, 'bar');
   });
 
   const expectDataSourceToBeBound = (
     app: AppWithRepoMixin,
-    ds: juggler.DataSource,
+    ds: Class<juggler.DataSource>,
```

Comment:
Does `Class<juggler.DataSource>` also allow instances of `juggler.DataSource`? I fee like it shouldn't, but if it does I think we should explicitly state so in a union type: `Class<juggler.DataSource> | juggler.DataSource`